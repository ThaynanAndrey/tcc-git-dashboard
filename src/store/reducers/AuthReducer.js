import { USER_LOGIN_SUCCESS, USER_LOGIN_ERROR, USER_LOGOUT_SUCCESS, USER_LOGOUT_ERROR } from '../actions/types';

export const initState = {
    user: undefined,
    accessToken: undefined,
    errorLogin: undefined,
    errorLogout: undefined
};
  
const authReducer = (state = initState, action) => {
    switch(action.type) {
        case USER_LOGIN_SUCCESS:
            return {
                ...state,
                user: action.user,
                idUser: action.idUser,
                accessToken: action.accessToken
            };
        case USER_LOGIN_ERROR:
            return {
                ...state,
                errorLogin: action.errorLogin
            }
        case USER_LOGOUT_SUCCESS:
            return initState;
        case USER_LOGOUT_ERROR:
            return {
                ...state,
                errorLogout: action.errorLogout
            }
        default:
          return state
    }
};
  
export default authReducer;