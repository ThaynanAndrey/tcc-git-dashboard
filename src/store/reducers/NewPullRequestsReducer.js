import { LOAD_PULL_REQUESTS_NO_PROJECT_SUCCESS, ADDED_PULL_REQUEST_SUCCESS, ADDED_PULL_REQUEST_ERROR } from '../actions/types';

const initState = {
    pullRequestsNoProject: []
};
  
const newPullRequestsReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PULL_REQUESTS_NO_PROJECT_SUCCESS:
            console.log('pull requests no project loaded!');
            return {
                ...state,
                pullRequestsNoProject: action.pullRequestsNoProject
            };
        case ADDED_PULL_REQUEST_SUCCESS:
            console.log("Pull request added");
            return state;
        case ADDED_PULL_REQUEST_ERROR:
            console.log("Error to add pull request");
            return state;
        default:
          return state
    }
};
  
export default newPullRequestsReducer;