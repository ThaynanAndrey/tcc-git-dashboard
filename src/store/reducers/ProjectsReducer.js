import { LOAD_PROJECTS_SUCCESS, LOAD_PROJECTS_ERROR, CREATE_PROJECT_SUCCESS,
    CREATE_PROJECT_ERROR, GET_PROJECT_SUCCESS } from '../actions/types';

export const initState = {
    projects: [],
    project: undefined,
    currentProject: undefined,
    error: undefined
};
  
const projectsReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PROJECTS_SUCCESS:
            return {
                ...state,
                projects: action.projects
            };
        case LOAD_PROJECTS_ERROR:
            return {
                ...state,
                projects: [],
                error: action.error
            };
        case CREATE_PROJECT_SUCCESS:
            return {
                ...state,
                projects: action.projects
            };
        case GET_PROJECT_SUCCESS:
            return {
                ...state,
                project: action.project
            }
        case CREATE_PROJECT_ERROR:
            return state;
        default:
          return state;
    }
};
  
export default projectsReducer;