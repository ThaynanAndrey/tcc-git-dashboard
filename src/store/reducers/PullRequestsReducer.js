import { LOAD_PROJECT_PULL_REQUESTS_SUCCESS, LOAD_PULL_REQUEST_SUCCESS, RESET_PROJECT_PULL_REQUESTS } from '../actions/types';

const initState = {
    projectPullRequests: [],
    selectedPullRequest: { responsavel: {}, repositorio: {} },
};
  
const pullRequestsReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PROJECT_PULL_REQUESTS_SUCCESS:
            console.log('get all pull requests from project!');
            return {
                ...state,
                projectPullRequests: action.projectPullRequests
            };
        case LOAD_PULL_REQUEST_SUCCESS:
            return {
                ...state,
                selectedPullRequest: action.selectedPullRequest
            }
        case RESET_PROJECT_PULL_REQUESTS:
            return {
                ...state,
                projectPullRequests: []
            }
        default:
          return state
    }
};
  
export default pullRequestsReducer;