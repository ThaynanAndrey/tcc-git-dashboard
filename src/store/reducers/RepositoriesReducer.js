import { LOAD_PROJECT_REPOSITORIES_SUCCESS, LOAD_REPOSITORIES_NO_PROJECT_SUCCESS, ADDED_REPOSITORY_SUCCESS,
    ADDED_REPOSITORY_ERROR, LOAD_EXTERNAL_REPOSITORIES, LOAD_EXTERNAL_REPOSITORIES_ERROR, RESET_PROJECT_REPOSITORIES,
    RESET_REPOSITORIES_NO_PROJECT } from '../actions/types';

export const initState = {
    projectRepositories: [],
    repositoriesNoProject: [],
    externalRepositories: [],
    msgExternalRepositories: undefined,
    error: undefined
};
  
const repositoriesReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PROJECT_REPOSITORIES_SUCCESS:
            return {
                ...state,
                projectRepositories: action.projectRepositories
            };
        case LOAD_REPOSITORIES_NO_PROJECT_SUCCESS:
            return {
                ...state,
                repositoriesNoProject: action.repositoriesNoProject
            }
        case ADDED_REPOSITORY_SUCCESS:
            return {
                ...state,
                repositoriesNoProject: action.repositoriesNoProject,
                externalRepositories: action.externalRepositories
            }
        case ADDED_REPOSITORY_ERROR:
            return {
                ...state,
                error: action.error
            }
        case LOAD_EXTERNAL_REPOSITORIES:
        case LOAD_EXTERNAL_REPOSITORIES_ERROR:
            return {
                ...state,
                externalRepositories: action.externalRepositories,
                msgExternalRepositories: action.msgExternalRepositories
            }
        case RESET_PROJECT_REPOSITORIES:
            return {
                ...state,
                projectRepositories: []
            }
        case RESET_REPOSITORIES_NO_PROJECT:
            return {
                ...state,
                repositoriesNoProject: [],
                externalRepositories: []
            }
        default:
          return state
    }
};
  
export default repositoriesReducer;