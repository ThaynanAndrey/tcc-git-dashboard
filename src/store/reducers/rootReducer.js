import { combineReducers } from 'redux';
import authReducer from './AuthReducer';
import pullRequestsReducer from './PullRequestsReducer';
import newPullRequestsReducer from './NewPullRequestsReducer';
import repositoriesReducer from './RepositoriesReducer';

const rootReducer = combineReducers({
    auth: authReducer,
    pullRequests: pullRequestsReducer,
    newPullRequests: newPullRequestsReducer,
    repositories: repositoriesReducer
});

export default rootReducer;