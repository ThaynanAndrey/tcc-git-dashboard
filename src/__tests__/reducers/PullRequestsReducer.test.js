import pullRequestsReducer from '../../store/reducers/PullRequestsReducer';
import { initState } from '../../store/reducers/PullRequestsReducer';

describe('PullRequestsReducer', () => {
    it('should return the initial state', () => {
        const state = pullRequestsReducer(undefined, {});
        
        expect(state).toMatchSnapshot();
        expect(state).toEqual(initState);
    });

    it('should handle LOAD_PROJECT_PULL_REQUESTS_SUCCESS', () => {
        const projectPullRequests = [ { name: "Old PR", date:"10/12/2015" } ];
        
        const action = {
            type: 'LOAD_PROJECT_PULL_REQUESTS_SUCCESS',
            projectPullRequests
        };
        
        const newState = {
            ...initState,
            projectPullRequests
        };

        const state = pullRequestsReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle LOAD_PULL_REQUEST_SUCCESS', () => {
        const selectedPullRequest = [ { name: "Old PR", date:"10/12/2015",
            responsavel: { name: "Joao" }, repositorio: { name: "GitRepo" } } ];
        
        const action = {
            type: 'LOAD_PULL_REQUEST_SUCCESS',
            selectedPullRequest
        };
        
        const newState = {
            ...initState,
            selectedPullRequest
        };

        const state = pullRequestsReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle RESET_PROJECT_PULL_REQUESTS', () => {
        const action = {
            type: 'RESET_PROJECT_PULL_REQUESTS'
        };
        
        const newState = {
            ...initState,
            projectPullRequests: []
        };

        const state = pullRequestsReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });
});