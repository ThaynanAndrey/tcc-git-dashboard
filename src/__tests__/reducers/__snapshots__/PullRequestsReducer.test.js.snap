// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PullRequestsReducer should handle LOAD_PROJECT_PULL_REQUESTS_SUCCESS 1`] = `
Object {
  "projectPullRequests": Array [
    Object {
      "date": "10/12/2015",
      "name": "Old PR",
    },
  ],
  "selectedPullRequest": Object {
    "repositorio": Object {},
    "responsavel": Object {},
  },
}
`;

exports[`PullRequestsReducer should handle LOAD_PULL_REQUEST_SUCCESS 1`] = `
Object {
  "projectPullRequests": Array [],
  "selectedPullRequest": Array [
    Object {
      "date": "10/12/2015",
      "name": "Old PR",
      "repositorio": Object {
        "name": "GitRepo",
      },
      "responsavel": Object {
        "name": "Joao",
      },
    },
  ],
}
`;

exports[`PullRequestsReducer should handle RESET_PROJECT_PULL_REQUESTS 1`] = `
Object {
  "projectPullRequests": Array [],
  "selectedPullRequest": Object {
    "repositorio": Object {},
    "responsavel": Object {},
  },
}
`;

exports[`PullRequestsReducer should return the initial state 1`] = `
Object {
  "projectPullRequests": Array [],
  "selectedPullRequest": Object {
    "repositorio": Object {},
    "responsavel": Object {},
  },
}
`;
